llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -1119,7 +1119,8 @@ static void unionWithMinMaxIntrinsicClamp(const IntrinsicInst *II,
                                           KnownBits &Known) {
   const APInt *CLow, *CHigh;
   if (isSignedMinMaxIntrinsicClamp(II, CLow, CHigh))
-    Known = Known.unionWith(ConstantRange(*CLow, *CHigh + 1).toKnownBits());
+    if (!(CLow->isMinSignedValue() && CHigh->isMaxSignedValue()))
```

Comment:
Instead of this check, you can use `ConstantRange::getNonEmpty` instead of the `ConstantRange` constructor. `getNonEmpty` will convert the range to a valid full set when Lower and Upper are equal.

Reply:
Right, thanks, corrected.